### PR TITLE
Add eCommerce signup flow

### PR DESCRIPTION
**Pull request by adamziel**
#### Changes proposed in this Pull Request

This PR is related to the new ecommerce plan, and as of now it's based on https://github.com/Automattic/wp-calypso/pull/28176. Once #28176 is merged, we'll be able to rebase it onto master to make the changes more clear. Also, we need to have the Blogger PR merged before we can add eCommerce plan to the "business" tab (https://github.com/Automattic/wp-calypso/pull/27125)

The purpose is to add a signup flow for the new plan. Specifically, this PR:
* Renames the `store-nux` signup flow to `ecommerce`
* Removes the `themes` step and defaults to twentyfifteen theme
* De-emphasises `.blog` domain suggestions
* Changes the plan recommendation to ecommerce
* Adds the ecommerce plan to plans list in signup and on /plans

#### TODO:

~~Once Blogger PR is merged, this needs to be rebased.~~ An update will be required for `client/my-sites/plans-features-main/index.jsx` to add the ecommerce plan to the "business plans" tab.

#### Testing instructions

* Create a new site
* Choose "Sell products or collect payments" as a primary goal in signup step 1
* You should be redirected to domains step, does not matter what you do you should never get a recommendation for `.blog` domain, only for other TLDs
* Once you choose a domain, you should be redirected to plan selection with one plan available - ecommerce plan
* Same test steps as in #27125, only with the eCommerce plan (once the TODO section above is addressed)

### CODE REVIEW

File: client/lib/plans/constants.js [hunk 1]
```diff
@@ -424,6 +434,59 @@ const getPlanBusinessDetails = () => ( {
 	getHiddenFeatures: () => [ FEATURE_AUDIO_UPLOADS ],
 } );
 
+const getPlanEcommerceDetails = () => ( {
+	group: GROUP_WPCOM,
+	type: TYPE_ECOMMERCE,
+	getTitle: () => i18n.translate( 'eCommerce' ),
+	getAudience: () => i18n.translate( 'Best for eCommerce sites' ),
+	getBlogAudience: () => i18n.translate( 'Best for online stores' ),
```
Comment (a8ci18n) [inline client/lib/plans/constants.js]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'Best for organizations and online stores.' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by zinigor** [comment]
I have deployed #28176 , rebased this one, should be good to go on.

**Comment by michaeldcain** [comment]
@zinigor: looks like it needs another rebase.

**Comment by retrofox** [comment]
Works as expected:

**Comment by jhnstn** [comment]
> You should never see a .wordpress.com free subdomain option
We no longer need that restriction now that we are using the staging domain to transfer sites.

**Comment by adamziel** [comment]
I'm back, will update this PR in a moment and hopefully we'll get it merged today

**Pull request merged**